docs/user_authentication/third_party_token_management.md

```diff
@@ -1,27 +1,25 @@
 
 ```sequence {theme: 'simple'}
 Title: Third party idp token management
-Pbench-Dashboard->Web-Server:[1] GET login page
-Web-Server->Pbench-Dashboard:[2] 200 login page
+Pbench-Dashboard->Pbench-Server:[1] GET login page
+Pbench-Server->Pbench-Dashboard:[2] 200 login page
 Pbench-Dashboard->Pbench-Dashboard:[3] Generate Authorization request
-Pbench-Dashboard->Web-Server:[4] 302 Redirect to identity provider
-Web-Server->Identity-Provider:[5] POST Authorization request
-Identity-Provider->Web-Server:[6] 200 Authenticate User
-Identity-Provider->Web-Server:[7] GET consent request
-Web-Server->Identity-Provider:[8] POST consent request payload
-Identity-Provider->Pbench-Dashboard:[9] 200 Authorization code
-Pbench-Dashboard->Identity-Provider:[10] POST Token request
-Identity-Provider->Pbench-Dashboard:[11] 200 Access/Refresh Token payload
-Pbench-Dashboard->Pbench-Server:[12] Send Access_token to get Pbench token
-Pbench-Server->Identity-Provider:[13] GET public Keys
-Identity-Provider->Pbench-Server:[14] 200 Public key
-Pbench-Server->Pbench-Server:[15] Validate the access token
-Pbench-Server->Identity-Provider:[16] [Optional] POST userinfo endpoint
-Pbench-Server->Pbench-Keycloak-server:[17] Exchange idp access token with Pbench token
-Pbench-Keycloak-server->Pbench-Server:[18] 200 new Pbench access_token and refresh_token
-Pbench-Server->Pbench-Dashboard:[19] 200 New Pbench token payload
-Pbench-Dashboard->Pbench-Server:[20] Request access to preotected endpoint
-Pbench-Server->Pbench-Server:[21] Validate Pbench token
-Pbench-Server->Pbench-Server:[22] Role extraction from token
-Pbench-Server->Pbench-Dashboard:[23] 200/401
+Pbench-Dashboard->Identity-Provider:[4] POST Authorization request
+Identity-Provider->Identity-Provider:[5] 200 Authenticate User
+Identity-Provider->Identity-Provider:[6] GET consent
+Identity-Provider->Pbench-Dashboard:[7] 200 Authorization code
+Pbench-Dashboard->Identity-Provider:[8] POST Token request
+Identity-Provider->Pbench-Dashboard:[9] 200 Access/Refresh Token payload
+Pbench-Dashboard->Pbench-Server:[10] Send Access_token to get Pbench token
+Pbench-Server->Identity-Provider:[11] GET public Keys
+Identity-Provider->Pbench-Server:[12] 200 Public key
+Pbench-Server->Pbench-Server:[13] Validate the access token
+Pbench-Server->Identity-Provider:[14] [Optional] POST userinfo endpoint
+Pbench-Server->Pbench-Keycloak-server:[15] Exchange idp access token with Pbench token
```

Comment:
No, it's an exchange because the identity provider is provided by our Keycloak server and we can configure it in such a way that that identity provider e.g Google would trust one of our confidential clients (and vice-versa) to share the tokens with each other. So 
1. user logs in with IDP and gets the access token
2. Presents the access token to the Pbench server and the Pbench server gives this to our internal confidential Keycloak client.
3. The client exchanges this token with a new token that originates from our Keycloak server. 
4. Usually token creation happens when the user provides his/her credentials but in this flow, the new user token generation happened without needing any credentials from the user. That's why its "exchange" , you can take a look at the documentation here [token_exchange](https://www.keycloak.org/docs/latest/securing_apps/#_token-exchange)

Reply:
Thanks for the link to the documentation!  I didn't realize that "exchange" was quite such a term-of-art.

If we wanted to use our own tokens, then having the (internal) Keycloak server exchange the external token for an internal one is pretty handy (e.g., I believe it saves us from having to implement token stuff in the Pbench Server).  But, I'm not sure that we want to add this level of complexity to our implementation...but, if it gets us roles and groups "for free", then it's probably worthwhile.